workbench-core/accounts/src/services/projectService.ts

```diff
@@ -80,4 +71,206 @@ export default class ProjectService {
 
     return Promise.resolve({ data: projectsResponse.Items as unknown as Project[] });
   }
+
+  /**
+   * Creates a new project
+   *
+   * @param params - the required fields to create a new project
+   * @param user - authenticated user creating the project
+   * @returns Project object of new project
+   */
+  public async createProject(params: CreateProjectReqeust, user: AuthenticatedUser): Promise<Project> {
+    // Verify caller is an IT Admin--TODO implement after dynamic AuthZ
+    // const userGroupsForCurrentUser: string[] = await this._dynamicAuthorizationService.getUserGroups(user.id);
+    // if(userGroupsForCurrentUser.length !== 1 || userGroupsForCurrentUser[0] !== 'ITAdmin') {
+    //   throw Boom.forbidden('Only IT Admins are allowed to create new Projects.');
+    // }
+
+    // Verify project name is unique and cost center exists
+    const resultsFromValidityChecks = await Promise.all([
+      this._isProjectNameInUse(params.name),
+      this._getCostCenter(params.costCenterId)
+    ]);
+    const costCenter: CostCenter = resultsFromValidityChecks[1];
+
+    // Generate Project ID
+    const projectId = uuidWithLowercasePrefix(resourceTypeToKey.project);
+
+    // Create ProjectAdmin and Researcher groups--TODO implement after dynamic AuthZ
+    // const createProjectAdminGroupResponse = this._dynamicAuthorizationService.createGroup(user, {
+    //   groupId: `${projId}#PA`,
+    //   description: `Project Admin group for ${projId}`
+    // });
+    // const createResearcherGroupResponse = this._dynamicAuthorizationService.createGroup(user, {
+    //   groupId: `${projId}#Researcher`,
+    //   description: `Researcher group for ${projId}`
+    // });
+    // if (!createProjectAdminGroupResponse.created || !createResearcherGroupResponse.created) {
+    //   throw Boom.badImplementation(
+    //     'Failed to create Project Admin group or Researcher group with dynamic authorization service.'
+    //   );
+    // }
+
+    // Create Permissions for the groups--TODO implement after dynamic AuthZ
+    // const identityPermissions: IdentityPermission[] = this._createIdentityPermissionsForProject(projectId);
+    // const createIdentityPermissionsResponse = this._dynamicAuthorizationService.createIdentityPermissions(
+    //   user,
+    //   identityPermissions
+    // );
+    // if (!createIdentityPermissionsResponse.created) {
+    //   throw Boom.badImplementation(
+    //     'Failed to create batch identity permissions for project with dyamic authorization service.'
+    //   );
+    // }
+
+    // Create Proj in DDB
+    const newProject: Project = {
+      id: projectId,
+      createdAt: new Date().toISOString(),
```

Comment:
We should set the current time as a variable and then set `createdAt` and `updatedAt` to be that variable. Otherwise, these two values will not match, when it should.
See here for an example
https://github.com/aws-solutions/solution-spark-on-aws/blob/develop/workbench-core/accounts/src/services/costCenterService.ts#L53

Reply:
Gotcha. Changing in line with that code example.